libraries/logging/stream-logger-android/src/main/java/io/getstream/logging/android/file/impl/LifecycleAwareLogFileManager.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.logging.android.file.impl
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Handler
+import android.os.Looper
+import android.widget.Toast
+import androidx.core.content.FileProvider
+import io.getstream.logging.android.file.StreamLogFileManager
+import io.getstream.logging.file.FileStreamLogger
+import java.io.File
+
+internal class LifecycleAwareLogFileManager(
+    private val fileLogger: FileStreamLogger,
+) : ActivityLifecycleCallbacks(), StreamLogFileManager.ShareManager, StreamLogFileManager.ClearManager {
+
+    private val handler = Handler(Looper.getMainLooper())
```

Comment:
Instead of been using thread, couldn't we use coroutines?

Reply:
It's a fully android specific class and the idea is to keep this class off 3rd party libraries as coroutines or rxjava and use android SDK only.